src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            enlargeCapacity();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
let’s also check the index - it should not be less than 0 or greater than size (it can be equal to size though)